File: Spark/python/word_count_rdd_gcp.py

from pyspark.sql import SparkSession

import re

spark = SparkSession.builder \
    .master('local[*]') \
    .appName('WordCount_RDD_GCP_Python') \
    .getOrCreate()
sc = spark.sparkContext

spark._jsc.hadoopConfiguration().set('google.cloud.auth.service.account.json.keyfile', '/home/to/your/credentials.json')

bucket_name = 'bucket_name'
file_name = '/textfile'
file_path = 'gs://' + bucket_name + file_name
rdd = sc.textFile(file_path)
top10 = rdd.flatMap(lambda item: item.split(' ')) \
    .map(lambda item: re.sub(r'[^a-zA-Z]', '', item).lower()) \
    .filter(lambda item: item != '') \
    .map(lambda item: (item, 1)) \
    .reduceByKey(lambda x, y: x + y) \
    .sortBy(lambda x: x[1], ascending=False) \
    .take(10)
print(top10)

File: Spark/scala/WordCountRDDGCP.scala

import org.apache.spark._
import org.apache.spark.rdd._

object WordCountRDDGCP {
    def main(args : Array[String]) : Unit => {
        val sparkConf = new SparkConf()
            .setMaster("local[*]")
            .setAppName("WordCount_RDD_GCP_Scala")
            //choose one from the belowing
            //export GOOGLE_APPLICATION_CREDENTIALS=/home/to/your/credentials.json
            .set("GOOGLE_APPLICATION_CREDENTIALS", "/home/to/your/credentials.json")
        val sc = spark.sparkContext
        
        val bucketName = args[0]
        val fileName = args[1]
        val filePath = "gs://" + bucketName + fileName
        val rdd = sc.textFile(filePath)
        val top10 = rdd.flatMap(_.split(' '))
            .map(_.replaceAll("[^a-zA-Z]", "").toLowerCase)
            .filter(_ != "")
            .map((_, 1))
            .reduceByKey(_ + _)
            .sortBy(_._2, ascending=false)
            .take(10)

        top10.foreach(println)
    }
}
